### PR TITLE
Specify the Size of List<T> Instances in InvokeTypeInfo

### PR DESCRIPTION
**Pull request by brianrob**
These allocations are showing up in some of the traces that I've been collecting recently.  Since the size of these `List<T>` instances is known, we can pre-allocate them.  This won't help for small instances, but for anything larger than 4, which is the current `DefaultSize`, this will eliminate resizing.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/InvokeTypeInfo.cs [hunk 1]
```diff
@@ -77,8 +77,8 @@ public override void WriteData(PropertyValue value)
         {
             if (this.properties != null)
             {
-                var membersNames = new List<string>();
-                var membersValues = new List<object?>();
+                var membersNames = new List<string>(this.properties.Length);
+                var membersValues = new List<object?>(this.properties.Length);
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/InvokeTypeInfo.cs:81]: Do these even need to be lists?  From a quick look at EventPayload, it seems these could just as easily be arrays?

Comment (brianrob) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/InvokeTypeInfo.cs]: That's a good point.  Yes, they can just be arrays.  I've pushed an update.

**Review by sywhang — APPROVED** [review]
LGTM

**Review by stephentoub — APPROVED** [review]

**Comment by brianrob** [comment]
Thanks for the reviews!

**Pull request merged**